firstvoices/backend/views/games_views.py

```diff
@@ -0,0 +1,118 @@
+    # inspired from previous fv-wordsy game
+    epoch = datetime(2024, 1, 1, 0, 0, 0).timestamp()  # wordsy game epoch
+    now = datetime.now().timestamp()
+    seconds_in_day = 86400000
+    index = math.floor((now - epoch) / seconds_in_day)
+    return index % num_words
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="Returns the wordsy config for the day.",
+        responses={
+            200: inline_serializer(
+                name="wordsy_response",
+                fields={
+                    "orthography": serializers.ListField(),
+                    "words": serializers.ListField(),
+                    "validGuesses": serializers.ListField(),
+                    "solution": serializers.CharField(),
+                },
+            ),
+            403: OpenApiResponse(description="Todo: Not authorized for this Site"),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+        parameters=[site_slug_parameter],
+    ),
+)
+class WordsyViewSet(SiteContentViewSetMixin, GenericViewSet):
+    http_method_names = ["get"]
+    pagination_class = None
+    queryset = ""
+
+    def list(self, request, **kwargs):
+        site = self.get_validated_site()[0]
+
+        # setting cache_expiry to 23:59 for current date
+        today = datetime.today().date()
+        eod = datetime.today().time().max
+        cache_expiry = (
+            datetime.combine(today, eod)
+            - datetime.combine(today, datetime.now().time())
+        ).total_seconds()
+
+        # Checking if required query sets are present in cache
+        cache_key_orthography = f"{site.title}-orthography"
+        cache_key_words = f"{site.title}-words"
+
+        # orthography
+        orthography_qs = caches[CACHE_KEY_WORDSY].get(cache_key_orthography)
+        if orthography_qs is None:
+            orthography_qs = list(
+                Character.objects.filter(site=site)
+                .order_by("sort_order")
+                .values_list("title", flat=True)
+            )
+            caches[CACHE_KEY_WORDSY].set(
+                cache_key_orthography, orthography_qs, cache_expiry
+            )
+
+        # words
+        words_qs = caches[CACHE_KEY_WORDSY].get(cache_key_words)
+        if words_qs is None:
+            # Filtering words based on their length excluding spaces
+            words_qs = list(
+                DictionaryEntry.objects.annotate(
+                    chars_length=RawSQL("ARRAY_LENGTH(split_chars_base, 1)", ())
+                )
+                .filter(
+                    site=site,
+                    visibility=site.visibility,
+                    exclude_from_games=False,
+                    exclude_from_kids=False,
+                    chars_length__exact=SOLUTION_LENGTH,
+                )
+                .exclude(title__contains=" ")
+                .order_by("custom_order")
+                .values_list("title", flat=True)
```

Comment:
I tested locally and found two issues with the resulting list:

1. The list should have only unique words - duplicates should be removed.
2. The words need to be provided to the game using base characters only (e.g. `split_chars_base` joined). This is because the titles might be using variations like uppercase, which won't be correctly processed by the game. (Example: alphabet `a b c` has variants `A B C`, and the dictionary has a word `Abcab`. this word should be in the config as `abcab` not `Abcab`.)